src/extensions.ts

```diff
@@ -313,6 +313,38 @@ export class SelectOriginByViwerCountry extends Custom {
   }
 };
 
+export interface AccessOriginByGeolocationProps {
+  /**
+   * The pre-defined country code table.
+   * Exampe: { 'US': 'amazon.com' }
+   */
+  readonly countryTable: { [code: string]: string };
+}
+
+/**
+ * Forward request to the nearest PoP as per geolocation.
+ */
+export class AccessOriginByGeolocation extends Custom {
+  constructor(scope: cdk.Construct, id: string, props: AccessOriginByGeolocationProps) {
+    const func = new NodejsFunction(scope, 'AccessOriginByGeolocationFunc', {
+      entry: `${EXTENSION_ASSETS_PATH}/cf-access-origin-by-geolocation/index.ts`,
+      // L@E does not support NODE14 so use NODE12 instead.
+      runtime: lambda.Runtime.NODEJS_12_X,
+      bundling: {
+        define: {
+          'process.env.COUNTRY_CODE_TABLE': jsonStringifiedBundlingDefinition(props.countryTable),
+        },
+      },
+    });
+    super(scope, id, {
+      func: func,
+      eventType: cf.LambdaEdgeEventType.ORIGIN_REQUEST,
+      solutionId: '',
```

Comment:
```suggestion
      solutionId: '8135',
```

Please use this solutionId thanks.

Reply:
SO8135 or 8135 ??

`DefaultDirIndex` use `SO8134`
https://github.com/pahud/cdk-cloudfront-plus/blob/d611a6b6167be7682418ac9277b61331ebdf4c55/src/extensions.ts#L256